### PR TITLE
Combine DTO and model layer

### PR DESCRIPTION
**Pull request by biafra23**
Handles #1053 

Changes proposed in this pull request:
- Remove DTO layer
- adjust tests

@gnosis/mobile-devs

### CODE REVIEW

File: data/src/main/java/io/gnosis/data/repositories/TokenRepository.kt [hunk 1]
```diff
@@ -23,7 +23,7 @@ class TokenRepository(
             else
```
Comment (biafra23) [inline data/src/main/java/io/gnosis/data/repositories/TokenRepository.kt:23]: This logic (replacing null token address with etherum info) should be implemented by the client gateway, shouldn't it?

File: data/src/main/java/io/gnosis/data/repositories/TokenRepository.kt [hunk 2]
```diff
@@ -23,7 +23,7 @@ class TokenRepository(
...
                 Balance(
                     TokenInfo(
```
Comment (elgatovital) [inline data/src/main/java/io/gnosis/data/repositories/TokenRepository.kt:25]: TokenType parsing should be done by adapter. Also this mapping step is not needed at all, as everything can be parsed directly from client gw response

Comment (biafra23) [reply data/src/main/java/io/gnosis/data/repositories/TokenRepository.kt:25]: It shouldn't be necessary. But we would need a custom adapter to implement this logic. I would rather have the client gateway send a response that doesn't need this logic. 

If you remove this code, tests will fail.

File: data/src/main/java/io/gnosis/data/repositories/TokenRepository.kt [hunk 3]
```diff
@@ -13,27 +16,10 @@ class TokenRepository(
...
         return CoinBalances(response.fiatTotal, response.items.map {
-
-            if (it.tokenInfo.address == null || it.tokenInfo.address == ZERO_ADDRESS)
-                Balance(
-                    ETH_TOKEN_INFO,
-                    it.balance,
-                    it.fiatBalance
-                )
+            if (it.tokenInfo.address == ZERO_ADDRESS)
+                it.copy(tokenInfo = it.tokenInfo.copy(logoUri = "local::ethereum"))
```
Comment (biafra23) [inline data/src/main/java/io/gnosis/data/repositories/TokenRepository.kt:20]: @jpalvarezl We should probably have this (logoUri for ETH transfers) in the client gateway?

**Review by elgatovital — APPROVED** [review]

**Pull request merged**